scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java

```diff
@@ -787,30 +789,34 @@ public void setTaskPausedOnError(InternalTask internalTask) {
         }
 
         Set<TaskId> updatedTasks = new HashSet<>();
-        for (InternalTask td : tasks.values()) {
+        for (InternalTask task : tasks.values()) {
             if (jobInfo.getStatus() == JobStatus.PENDING) {
-                td.setStatus(TaskStatus.SUBMITTED);
-                updatedTasks.add(td.getId());
+                task.setStatus(TaskStatus.SUBMITTED);
+                updatedTasks.add(task.getId());
             } else if ((jobInfo.getStatus() == JobStatus.RUNNING) ||
                 (jobInfo.getStatus() == JobStatus.STALLED)) {
-                if ((td.getStatus() != TaskStatus.FINISHED) && (td.getStatus() != TaskStatus.RUNNING) &&
-                    (td.getStatus() != TaskStatus.SKIPPED) && (td.getStatus() != TaskStatus.FAULTY)
-                        && (td.getStatus() != TaskStatus.PAUSED_ON_ERROR)) {
-                    td.setStatus(TaskStatus.PENDING);
-                    updatedTasks.add(td.getId());
+                if ((task.getStatus() != TaskStatus.FINISHED) && (task.getStatus() != TaskStatus.RUNNING) &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)